src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.cs

```diff
@@ -154,5 +154,75 @@ public void IReadOnlyDictionary_Generic_Values_ContainsAllCorrectValues(int coun
         }
 
         #endregion
+
+        #region Remove(TKey)
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_RemoveKey_ValidKeyNotContainedInDictionary(int count)
+        {
+            Dictionary<TKey, TValue> dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+            TValue value;
+            TKey missingKey = GetNewKey(dictionary);
+
+            Assert.False(dictionary.Remove(missingKey, out value));
+            Assert.Equal(count, dictionary.Count);
+            Assert.Equal(default(TValue), value);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_RemoveKey_ValidKeyContainedInDictionary(int count)
+        {
+            Dictionary<TKey, TValue> dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+            TKey missingKey = GetNewKey(dictionary);
+            TValue value;
+
+            dictionary.Add(missingKey, CreateTValue(34251));
+            Assert.True(dictionary.Remove(missingKey, out value));
+            Assert.Equal(count, dictionary.Count);
+            Assert.Equal(CreateTValue(34251), value);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_RemoveKey_DefaultKeyNotContainedInDictionary(int count)
+        {
+            Dictionary<TKey, TValue> dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+            TValue value = CreateTValue(12345);
+
+            if (DefaultValueAllowed)
+            {
+                TKey missingKey = default(TKey);
+                while (dictionary.ContainsKey(missingKey))
+                    dictionary.Remove(missingKey);
+                Assert.False(dictionary.Remove(missingKey, out value));
+                Assert.Equal(default(TValue), value);
+            }
+            else
+            {
+                Assert.Throws<ArgumentNullException>(() => dictionary.Remove(default(TKey)));
+                Assert.Equal(CreateTValue(12345), value);
```

Comment:
Oops... The lines 204 and 205 were supposed to be like this...
```C#
Assert.Throws<ArgumentNullException>(() => dictionary.Remove(default(TKey), out outValue));
Assert.Equal(refValue, outValue);
```

Where `outValue` was supposed to be assigned with `refValue` and after the exception the assert was to ensure that the `outValue` hasn't changed from `refValue`. This is based on the knowledge that the `Remove(K, V)` method (just like existing `Remove(K)` method) doesn't assign `default(TValue)` to output parameter in the Null Key case.

Changed the code...